compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -15,6 +15,46 @@
 
 namespace mlir::iree_compiler::AMDAIE {
 
+/// Sets the lowering configuration for a generic op implementing a
+/// transposition.
+static LogicalResult setTransposeLikeOpRootConfig(
```

Comment:
This is confusing. I think you need to use something like `ContractionOpInterface` (https://github.com/llvm/llvm-project/blob/25cec33521ba1c67f5e939ceea277be18b2d0fb7/mlir/include/mlir/Dialect/Linalg/IR/LinalgInterfaces.td#L21) and check if the op is a "matmul" of the form you need. You might need to add some checkers similar to what the interface has to check for matmul/batch_matmul, etc.

You dont need to change the interface upstream to start with. You can just start with writing a similar matcher here and set the config using that. You can then graduate the matcher upstream.